lib/service/Service.ts

```diff
@@ -139,22 +138,52 @@ class Service extends EventEmitter {
   }
 
   /**
-   * Get a list of standing orders from the order book.
+   * Get a list of standing peer orders from the order book for a specified trading pair, or for all
+   * trading pairs if no pair is specified.
    */
-  public getOrders = ({ pairId, maxResults }: { pairId: string, maxResults: number }) => {
-    return this.orderBook.getPeerOrders(pairId, maxResults);
+  public getOrders = ({ pairId, maxResults }: { pairId?: string, maxResults?: number }) => {
+    const ret: OrderArrays = {
+      buyOrders: [],
+      sellOrders: [],
+    };
+    if (pairId) {
+      const orderArrays = this.orderBook.getPeerOrders(pairId, maxResults ? maxResults : 0);
+      ret.buyOrders.concat(orderArrays.buyOrders);
+      ret.sellOrders.concat(orderArrays.sellOrders);
+    } else {
+      this.orderBook.pairs.forEach((pair) => {
+        const orderArrays = this.orderBook.getPeerOrders(pair.id, maxResults ? maxResults : 0);
+        ret.buyOrders.concat(orderArrays.buyOrders);
+        ret.sellOrders.concat(orderArrays.sellOrders);
+      });
+    }
+    return ret;
   }
 
   /**
    * Add an order to the order book.
    */
   public placeOrder = async (order: OwnOrder) => {
-    assert(order.price >= 0, 'price cannot be negative');
-    assert(order.quantity !== 0, 'quantity must not equal 0');
-    if (order.price === 0) {
-      return this.orderBook.addMarketOrder(order);
-    } else {
-      return this.orderBook.addLimitOrder(order);
+    if (order.price < 0) {
+      throw {
+        message: 'price cannot be negative',
+        code: status.INVALID_ARGUMENT,
```

Comment:
Sorry for being the separation of concerns police, but `Service` shouldn't know/use anything from `gRPC`.

Reply:
That's fine, I just wasn't seeing an easy way to do this without `Service` at least knowing the error codes. Do you have an alternative to suggest? I guess I could put all error handling in `GrpcService.unaryCall` and then create error types for these invalid argument errors - or maybe go back to AssertionErrors and always treat those as `INVALID_ARGUMENT`.

Reply:
simplest solution would be to create a list of errors for `Service` which would be the equivalent to these of gRPC you wanted to use. `GrpcService` will catch them, and link them to the gRPC errors. This way we can continue to expand how we want to expose internal errors to the gRPC client. At the moment we have no support for that. 

It might be what you suggested, but I wasn't sure.